### PR TITLE
TST: collect indexing tests by method

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jbrockmendel** [comment]
@jreback gentle ping for this and #39873, blockers for more test sharing

File: pandas/tests/indexing/multiindex/test_setitem.py [hunk 1]
```diff
@@ -15,114 +15,109 @@
 import pandas.core.common as com
 
 
+def assert_equal(a, b):
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_setitem.py:18]: this is shadowing a common function, would not do this

Comment (jreback) [reply pandas/tests/indexing/multiindex/test_setitem.py:18]: you use a lambda

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
ping on green

File: pandas/tests/indexing/multiindex/test_setitem.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/indexing/multiindex/test_setitem.py:18]: it already exists, just moving it to a shared namespace

Comment (jreback) [reply pandas/tests/indexing/multiindex/test_setitem.py:18]: right but i think should change it here

Comment (jbrockmendel) [reply pandas/tests/indexing/multiindex/test_setitem.py:18]: sure.  follow-up?  the CI is gonna be slammed today

**Review by jreback — APPROVED** [review]

**Pull request merged**